lib/Graph/Nodes.cpp

```diff
@@ -1290,32 +1290,30 @@ bool GatherRangesNode::verify() const {
   return isValid;
 }
 
-bool ScatterAssignNode::verify() const {
+bool ScatterDataNode::verify() const {
```

Comment:
Can you add a check here that data and slices should have the same element kind?